packages/lisk-transactions/src/utils/validate_multisignatures.ts

```diff
@@ -0,0 +1,54 @@
+import * as cryptography from '@liskhq/lisk-cryptography';
+import { Account, TransactionJSON } from '../transaction_types';
+
+export const validateMultisignatures = (
+	sender: Account,
+	transaction: TransactionJSON,
+	transactionHash: Buffer,
+): ReadonlyArray<string> => {
+	const minimumValidations = sender.multimin || 0;
+	const transactionSignatures = transaction.signatures as ReadonlyArray<string>;
+	const senderPublicKeys = sender.multisignatures || [];
+	// tslint:disable-next-line no-let
+	let validatedSignaturesCount = 0;
+	// tslint:disable-next-line no-let
+	let checkedPublicKeys: ReadonlyArray<string> = [];
+
+	const unvalidatedSignatures = senderPublicKeys.reduce(
+		(
+			signaturesArray: ReadonlyArray<string>,
+			publicKey: string,
+		): ReadonlyArray<string> => {
+			if (checkedPublicKeys.includes(publicKey)) {
+				return signaturesArray;
+			}
+
+			return [
+				...signaturesArray,
+				...transactionSignatures.reduce(
+					(
+						collectedSignatures: ReadonlyArray<string>,
+						signature: string,
+					): ReadonlyArray<string> => {
+						if (
+							cryptography.verifyData(transactionHash, signature, publicKey)
+						) {
+							checkedPublicKeys = [publicKey, ...checkedPublicKeys];
+							validatedSignaturesCount++;
+
+							return collectedSignatures;
+						}
+
+						return [signature, ...collectedSignatures];
+					},
+					[],
+				),
+			];
+		},
+		[],
+	);
+
+	return validatedSignaturesCount >= minimumValidations
```

Comment:
This function will never return an error? Even if we have some invalid signatures in the transaction.signatures array. Is that the expected behavior?

Reply:
I can return errors directly from here. not sure if we should also return errors if there were more valid signatures than the minimum amount.